rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -257,14 +262,11 @@ object ProcNormalizeMatcher {
                           p.proc_2,
                           ProcVisitInputs(VectorPar(), input.env, leftResult.knownFree))
           lp = leftResult.par
-          resultConnective = if (lp.sends.isEmpty && lp.receives.isEmpty && lp.news.isEmpty && lp.exprs.isEmpty && lp.matches.isEmpty && lp.bundles.isEmpty) {
-            lp.connectives match {
-              case List(Connective(ConnAndBody(ConnectiveBody(ps)))) =>
-                Connective(ConnAndBody(ConnectiveBody(ps :+ rightResult.par)))
-              case _ => Connective(ConnAndBody(ConnectiveBody(Vector(lp, rightResult.par))))
-            }
-          } else {
-            Connective(ConnAndBody(ConnectiveBody(Vector(lp, rightResult.par))))
+          resultConnective = lp.onlyConnectives() match {
```

Comment:
I've extracted the `if(…)` condition to [helper function](https://github.com/goral09/rchain/blob/29f7a6a60d18bbbcd88c2484057c9f60da9d317d/models/src/main/scala/coop/rchain/models/rholang/implicits.scala#L250)

Reply:
Really, the helper function should be looking for singleton lists, like the other helpers.